### PR TITLE
Legg til endepunkt for opprettelse av typer

### LINKED ISSUE #14: det må lages eit post-endepunkt på / som gjer klienten i stand til å opprette ontologi-typer.
Samtidig må det
lages eit post-endepunkt på / som gjer klienten i stand til å opprette ontologi-typer.

Tas ut i eget issue.

_Originally posted by @stigbd in https://github.com/Informasjonsforvaltning/static-rdf-server/issues/13#issuecomment-1034665198_

### PR DESCRIPTION
**Pull request by stigbd**
Fixes #14

Denne PR retter også opp i retur-kode ved PUT:
- 201 Created når ressursen ikkje fins
- 204 No Content når ressursen fins

~~I begge tilfeller returneres location header~~
Location header returneres ved 201

### CODE REVIEW

File: static_rdf_server/routes/ontology.py [hunk 1]
```diff
@@ -100,7 +107,8 @@ async def put_ontology(request: web.Request) -> web.Response:  # noqa: C901
         with open(path, "w") as file:
             file.write(ontology_file)
 
-    return web.Response(status=204)
+    headers = MultiDict([(hdrs.LOCATION, f"{ontology_type}/{ontology}")])
```
Comment (jeffreiffers) [inline static_rdf_server/routes/ontology.py:110]: Url skal være absolutt.

The Location response-header field is used to redirect the recipient to a location other than the Request-URI for completion of the request or identification of a new resource. For 201 (Created) responses, the Location is that of the new resource which was created by the request. For 3xx responses, the location SHOULD indicate the server's preferred URI for automatic redirection to the resource. The field value consists of a single absolute URI.

**Review by jeffreiffers — CHANGES_REQUESTED** [review]

File: tests/integration/test_put_ontology.py [hunk 2]
```diff
@@ -58,20 +58,19 @@ async def test_put_ontology_when_ontology_does_not_exist(client: Any, fs: Any) -
     response = await client.put(
         f"/{ontology_type}/{ontology}", headers=headers, data=mpwriter
     )
-    if response.status != 204:
-        body = await response.json()
 
-    assert response.status == 204, body
+    assert response.status == 201
+    assert f"{ontology_type}/{ontology}" in response.headers[hdrs.LOCATION]
```
Comment (jeffreiffers) [inline tests/integration/test_put_ontology.py:63]: Absolutt...

File: static_rdf_server/routes/ontology.py (hunk 1 above)
Comment (stigbd) [reply static_rdf_server/routes/ontology.py:110]: Ikkje iflg https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Location#directives

File: tests/integration/test_put_ontology.py [hunk 3]
```diff
@@ -118,10 +117,9 @@ async def test_put_ontology_when_ontology_does_exist(client: Any, fs: Any) -> No
     response = await client.put(
         f"/{ontology_type}/{ontology}", headers=headers, data=mpwriter
     )
-    if response.status != 204:
-        body = await response.json()
 
-    assert response.status == 204, body
+    assert response.status == 204
+    assert f"{ontology_type}/{ontology}" in response.headers[hdrs.LOCATION]
```
[the remark this answers is not in the document]
Comment (stigbd) [reply tests/integration/test_put_ontology.py]: Sjå over.

File: tests/integration/test_put_ontology.py (hunk 2 above)
Comment (stigbd) [reply tests/integration/test_put_ontology.py:63]: Sjå over.

**Pull request merged**